### PR TITLE
chore: Revert "feat: generate SDKs for Looker 23.18"

### PR DESCRIPTION
**Pull request by drstrangelooker**
Reverts looker-open-source/sdk-codegen#1378

This allows other code to land while waiting for sdk_23_18 without 3.1 endpoints.

### CODE REVIEW

File: csharp/rtl/Constants.cs [hunk 1]
```diff
@@ -61,7 +61,7 @@ public struct Constants
 
         public const string DefaultApiVersion = "4.0";
         public const string AgentPrefix = "CS-SDK";
-        public const string LookerVersion = "23.18";
+        public const string LookerVersion = "23.17";
```
Comment (jkaster) [inline csharp/rtl/Constants.cs:64]: This isn't based on a CI image then? Should it be 23.16 still?

File: kotlin/src/main/com/looker/sdk/Constants.kt [hunk 2]
```diff
@@ -28,7 +28,7 @@ package com.looker.sdk
 
 const val ENVIRONMENT_PREFIX = "LOOKERSDK"
 const val SDK_TAG = "KT-SDK"
-const val LOOKER_VERSION = "23.18"
+const val LOOKER_VERSION = "23.17"
```
Comment (jkaster) [inline kotlin/src/main/com/looker/sdk/Constants.kt:31]: consistent with the bad version. If this isn't going to be published I'm not sure it matters.

File: packages/sdk/src/4.0/funcs.ts [hunk 3]
```diff
@@ -6805,13 +6811,13 @@ export const search_groups = async (
...
  * @param sdk IAPIMethods implementation
- * @param request composed interface "IRequestSearchGroups" for complex method parameters
+ * @param request composed interface "IRequestSearchGroupsWithRoles" for complex method parameters
  * @param options one-time API call overrides
  *
  */
 export const search_groups_with_roles = async (
   sdk: IAPIMethods,
-  request: IRequestSearchGroups,
+  request: IRequestSearchGroupsWithRoles,
```
Comment (jkaster) [inline packages/sdk/src/4.0/funcs.ts:6820]: interesting. I wonder if we've introduced a breaking change in this SDK call.

**Review by jkaster — COMMENTED** [review]
Not sure about the 23.17 version string

File: csharp/rtl/Constants.cs (hunk 1 above)
Comment (drstrangelooker) [reply csharp/rtl/Constants.cs:64]: No, somehow the version number got screwed up last time. But when we publish this again they'll be fixed.

File: packages/sdk/src/4.0/funcs.ts (hunk 3 above)
Comment (drstrangelooker) [reply packages/sdk/src/4.0/funcs.ts:6820]: Not sure.

Comment (drstrangelooker) [reply packages/sdk/src/4.0/funcs.ts:6820]: This is just a revert so we will address when the sdk_23_18 actually rolls forward.

**Review by jkaster — APPROVED** [review]

**Pull request merged**